src/test/java/io/confluent/connect/elasticsearch/ElasticsearchClientTest.java

```diff
@@ -565,6 +566,102 @@ public void testNoVersionConflict() throws Exception {
     client2.close();
   }
 
+  @Test
+  public void testIncompatibleESVersionDataStreamNotSet() throws Exception {
+    container.close();
+    container = new ElasticsearchContainer(DEFAULT_DOCKER_IMAGE_NAME + ":" + "7.8.1");
+    container.start();
+
+    String address = container.getConnectionUrl().replace(container.getContainerIpAddress(), container.hostMachineIpAddress());
+    props.put(CONNECTION_URL_CONFIG, address);
+    props.put(CONNECTION_USERNAME_CONFIG, "elastic");
+    props.put(CONNECTION_PASSWORD_CONFIG, ELASTIC_PASSWORD);
+    config = new ElasticsearchSinkConnectorConfig(props);
+    converter = new DataConverter(config);
+
+    ElasticsearchClient client = new ElasticsearchClient(config, null);
+    helperClient = new ElasticsearchHelperClient(address, config);
+    client.createIndexOrDataStream(index);
+
+    writeRecord(sinkRecord(0), client);
+    client.flush();
+
+    waitUntilRecordsInES(1);
+    assertEquals(1, helperClient.getDocCount(index));
+    client.close();
+    helperClient = null;
+
+    container.close();
+    container = ElasticsearchContainer.fromSystemProperties();
+    container.start();
+  }
+
+  @Test
+  public void testCompatibleESVersionDataStreamNotSet() throws Exception {
+    container.close();
+    String ESVersion = "7.9.3";
+    container = new ElasticsearchContainer(DEFAULT_DOCKER_IMAGE_NAME + ":" + ESVersion);
```

Comment:
let's add a constructor that just takes the version number and constructs the docker image name inside